opentech/apply/utils/blocks.py

```diff
@@ -0,0 +1,163 @@
+from collections import Counter
+
+import bleach
+from django.core.exceptions import ValidationError
+from django.forms.utils import ErrorList
+from django.utils.translation import ugettext_lazy as _
+from django.utils.text import mark_safe
+
+from wagtail.core.blocks import StaticBlock, StreamValue, StreamBlock
+
+from tinymce.widgets import TinyMCE
+
+from opentech.apply.stream_forms.blocks import FormFieldBlock, TextFieldBlock
+
+
+def find_duplicates(items):
+    counted = Counter(items)
+    duplicates = [
+        name for name, count in counted.items() if count > 1
+    ]
+    return duplicates
+
+
+def prettify_names(sequence):
+    return [nice_field_name(item) for item in sequence]
+
+
+def nice_field_name(name):
+    return name.title().replace('_', ' ')
+
+
+class RichTextFieldBlock(TextFieldBlock):
+    widget = TinyMCE(mce_attrs={
+        'elementpath': False,
+        'branding': False,
+        'toolbar1': 'undo redo | styleselect | bold italic | bullist numlist | link',
+        'style_formats': [
+            {'title': 'Headers', 'items': [
+                {'title': 'Header 1', 'format': 'h1'},
+                {'title': 'Header 2', 'format': 'h2'},
+                {'title': 'Header 3', 'format': 'h3'},
+            ]},
+            {'title': 'Inline', 'items': [
+                {'title': 'Bold', 'icon': 'bold', 'format': 'bold'},
+                {'title': 'Italic', 'icon': 'italic', 'format': 'italic'},
+                {'title': 'Underline', 'icon': 'underline', 'format': 'underline'},
+            ]},
+        ],
+    })
+
+    class Meta:
+        label = _('Rich text field')
+        icon = 'form'
+
+    def get_searchable_content(self, value, data):
+        return bleach.clean(data, tags=[], strip=True)
+
+
+class CustomFormFieldsBlock(StreamBlock):
+    rich_text = RichTextFieldBlock(group=_('Fields'))
+    required_blocks = None
```

Comment:
`required_blocks = None` should be an interable by default e.g. `[]` otherwise `[block.name for block in self.required_blocks]` will error if you dont define it.